docs/places.mdown

```diff
@@ -0,0 +1,107 @@
+* A storage layer that manages database connections and provides convenience methods for executing SQL statements.
+* A [foreign function interface](https://docs.rs/ffi-support) (FFI) that wraps API and storage calls for Swift and Kotlin consumers.
+
+History
+=======
+
+The history API provides methods for managing URLs, autocompleting address bar entries, and recording and clearing history visits.
+
+## Recording visits
+
+`history::apply_observation` records a visit observation for a URL. A visit observation notes the page title and URL, and whether the visit was from following a link, address bar entry, bookmark, or redirect.
+
+`history::update_frecency` recalculates frecency for a URL. Frecency is a score, based on the _fre_quency and _recency_ of visits, used to rank pages in autocomplete results. The frecency algorithm is an implementation detail, and subject to [change](https://github.com/mozilla/application-services/issues/610), so raw scores are not meaningful on their own.
+
+## Autocomplete matching
+
+`history_api::search_frecent` returns all matches by origin, URL fragment, past autocomplete input, title, and tags, ranked by frecency and relevance.
+
+`history_api::match_url` returns the highest-ranked autocomplete match for an origin or URL fragment.
+
+`history_api::accept_result` records an input that completed to the chosen URL. This is stored and used in `search_frecent` to return matches by past autocomplete input.
+
+## Clearing history
+
+`history::delete_place_by_guid` deletes a URL and all its visits, and writes a Sync tombstone if the URL was synced. Bookmarked URLs can't be deleted until the bookmark is either deleted, or changed to point to a different URL.
+
+`history::delete_visits_between` deletes visits to a URL in a time range. If all visits to the URL fall within the range, the URL is also removed and tombstoned, except if it's bookmarked.
+
+`history::delete_place_visit_at_time` deletes a visit to a URL at a specific time. If it's the last remaining visit, the URL is also removed and tombstoned, except if it's bookmarked.
+
+`history::prune_destructively` prunes old history visits, without writing tombstones or Sync metadata. It's recommended to call this as a last resort—for example, if the device is low on disk space. Old visits _may_ be resurrected through Sync; however, since we only sync a subset of history, and expire old synced history after 60 days, this may cause data loss.
+
+`history::wipe_local` erases all history, without writing tombstones or Sync metadata. This means that visits from other devices can reappear over time, and a full sync is likely to resurrect all cleared history. `wipe_local` is meant for cases like clearing local data when signing out of Sync.
+
+Bookmarks
+=========
+
+The bookmarks API provides methods for creating, organizing, and deleting bookmark items: bookmarks, folders, and separators.
+
+## Creating bookmarks
```

Comment:
I worry that these are too specific, and we could easily fail to update this doc when changing details about the methods. E.g. I'm not sure the value in putting them here outweighs the cost of updating them and the likelihood that they'll get out of date. Ditto for history, but the bookmarks are less filled out, so.

We also would want to be clear that this isn't really how the public (swift/kotlin) APIs are structured.

Reply:
I agree, I think it would be more maintainable if we could link out to some nice auto-generated-from-comments API docs for this section, but still provide a higher-level overview of the concepts needed to understand those docs.